pandas/tests/frame/test_indexing.py

```diff
@@ -2626,6 +2626,20 @@ def test_index_namedtuple(self):
         result = df.loc[IndexType("foo", "bar")]["A"]
         assert result == 1
 
+    def test_index_single_double_tuples(self):
+        # GH 20991
+        tuple_1 = tuple([1, 2])
+        tuple_2 = tuple([1])
+        df = DataFrame([[tuple_1], [tuple_2]], columns=["A"]).set_index("A")
```

Comment:
Can you just construct `idx = pd.Index(...)` before this and use `index=idx` as part of the DataFrame constructors? Would be more concise than the current construction / set_index ops

Reply:
Thanks @WillAyd - removed `.set_index()`